### PR TITLE
table/tables: make CI TestCacheTableBasicReadAndWrite more stable

### LINKED ISSUE #30922: Unstable test TestCacheTableBasicReadAndWrite
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Found in CI
https://ci.pingcap.net/blue/rest/organizations/jenkins/pipelines/tidb_ghpr_unit_test/runs/26002/nodes/52/steps/57/log/?start=0

```
[2021-12-20T08:11:05.302Z] --- FAIL: TestCacheTableBasicReadAndWrite (0.33s)
[2021-12-20T08:11:05.302Z]     cache_test.go:225: 
[2021-12-20T08:11:05.302Z]         	Error Trace:	cache_test.go:225
[2021-12-20T08:11:05.302Z]         	Error:      	Should be true
[2021-12-20T08:11:05.302Z]         	Test:       	TestCacheTableBasicReadAndWrite
```

### 2. What did you expect to see? (Required)

CI stable

### 3. What did you see instead (Required)

Some time it fails

### 4. What is your TiDB version? (Required)

Master

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #30922

Problem Summary:

### What is changed and how it works?

Make the CI `TestCacheTableBasicReadAndWrite ` more stable.

- sleep a while to wait the table cache loading to make the test more stable
- avoid repeating spawn read lock goroutine when the previous one is still ongoing

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: table/tables/cache_test.go [hunk 1]
```diff
@@ -221,6 +221,7 @@ func TestCacheTableBasicReadAndWrite(t *testing.T) {
 		if tk.HasPlan("select * from write_tmp1", "UnionScan") {
 			break
 		}
+		time.Sleep(50 * time.Millisecond)
```
Comment (bb7133) [inline table/tables/cache_test.go:224]: Please add some comments on why adding the `Sleep` here.

Comment (tiancaiamao) [reply table/tables/cache_test.go:225]: Done https://github.com/pingcap/tidb/pull/30924/files#diff-aa5d7f1cfcba5189e1a710fe7b29f53a8a5ef4ed9877af8e666fc66efe6db218R224

Comment (lcwangchao) [reply table/tables/cache_test.go:225]: Why wait for the cache to be loaded. I think when we got the UnionScan, the cache is already loaded.

Comment (lcwangchao) [reply table/tables/cache_test.go:225]: Did not see the break above... Ignore it...

**Review by lcwangchao — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**